Problem analysis (#28)
chhoumann/p7: report/sections/Chapter 1/MVP.tex

\section{Minimal Viable Product}
Considering that students use a variety of different operating systems, a web application would be the best way to ensure easy portability.
To facilitate the problems mentioned in \ref{sec:initial-problem}, we imagine a website that allows users to complete the following tasks:
\begin{itemize}

> 	\item The professor should be able to create a syllabus with exercises that students are expected to solve

Reviewer: ```suggestion
	\item The professor should be able to create a syllabus with exercises that students are expected to solve.
```

	\begin{itemize}

> 		\item The professor should be able to create multiple syllabi where students can self-enroll

Reviewer: ```suggestion
		\item The professor should be able to create multiple syllabi where students can self-enroll.
```

	\end{itemize}

> 	\item The professor should be able to track the amount of exercises solved by each enrolled student

Reviewer: ```suggestion
	\item The professor should be able to track the amount of exercises solved by each enrolled student.
```

	\begin{itemize}

> 		\item The professor should be able to view additional statistics like the number of attempts used by a student to solve an exercise

Reviewer: ```suggestion
		\item The professor should be able to view additional statistics like the number of attempts used by a student to solve an exercise.
```

> 		\item The professor should be able to view additional statistics like the number of attempts used by a student to solve an exercise

Reviewer: ```suggestion
		\item The professor should be able to view additional statistics like the number of attempts used by a student to solve an exercise.
```

	\end{itemize}

> 	\item The professor should be able to specify criteria in order to pass each exercise 

Reviewer: ```suggestion
	\item The professor should be able to specify criteria in order to pass each exercise.
```

	\begin{itemize}

> 		\item Students should be able to verify whether their solution is correct

Reviewer: ```suggestion
		\item Students should be able to verify whether their solution is correct.
```

> 		\item Verifying solutions should be automatically handled by the system

Reviewer: ```suggestion
		\item Verifying solutions should be automatically handled by the system.
```

	\end{itemize}

> 	\item Students should be able to collaborate on the same code

Reviewer: ```suggestion
	\item Students should be able to collaborate on the same code.
```

> 	\item Students should have access to discussed solutions later on

Reviewer: ```suggestion
	\item Students should have access to discussed solutions later on.
```

> 	\item The professor should be able to see each student's code

Reviewer: ```suggestion
	\item The professor should be able to see each student's code.
```

\end{itemize}

We then prioritized the presented features required to construct a minimal viable product (MVP).

> We plan on using this in practice to see whether it serves as a useful solution to the presented problems. 

Reviewer: ```suggestion
We intend to present this MVP as a prototype to potential users in order to determine its usability.
```

> The MVP is meant to be used as a demo product for gathering additional feedback.

Reviewer: ```suggestion

```

> As such, only a minimal number of features are included in our proposed solution:

Reviewer: Will this be our proposed solution or our MVP? Maybe we should clarify this.
